src/kakarot/instructions/stop_and_math_operations.cairo

```diff
@@ -471,15 +471,23 @@ namespace StopAndMathOperations {
         let popped = cast([ap - 1], Uint256*);
 
         // compute y = (x >> (248 - i * 8)) & 0xFF
-        let (mul, _) = uint256_mul(popped[0], Uint256(8, 0));
-        let (right) = uint256_sub(Uint256(248, 0), mul);
-        let (shift_right) = uint256_shr(popped[1], right);
-        let (result) = uint256_and(shift_right, Uint256(0xFF, 0));
-
-        tempvar bitwise_ptr = bitwise_ptr;
-        tempvar range_check_ptr = range_check_ptr;
-        tempvar result = Uint256(result.low, result.high);
-        jmp end;
+        let i = popped[0];
+        let (is_inf_32) = uint256_lt(i, Uint256(32, 0));
+        if (is_inf_32 != FALSE) {
+            let (mul, _) = uint256_mul(i, Uint256(8, 0));
+            let (right) = uint256_sub(Uint256(248, 0), mul);
+            let (shift_right) = uint256_shr(popped[1], right);
+            let (result) = uint256_and(shift_right, Uint256(0xFF, 0));
+            tempvar bitwise_ptr = bitwise_ptr;
+            tempvar range_check_ptr = range_check_ptr;
+            tempvar result = Uint256(result.low, result.high);
+            jmp end;
+        } else {
+            tempvar bitwise_ptr = bitwise_ptr;
+            tempvar range_check_ptr = range_check_ptr;
+            tempvar result = Uint256(0, 0);
+            jmp end;
+        }
```

Comment:
I don't really like to mix if/else and low level jmp

```suggestion
        let i = popped[0];
        let (is_inf_32) = uint256_lt(i, Uint256(32, 0));
        jmp byte_i if is_inf_32 != 0;
        
        tempvar bitwise_ptr = bitwise_ptr;
        tempvar range_check_ptr = range_check_ptr;
        tempvar result = Uint256(0, 0);
        jmp end if ;
        
        byte_i:
        let (mul, _) = uint256_mul(i, Uint256(8, 0));
        let (right) = uint256_sub(Uint256(248, 0), mul);
        let (shift_right) = uint256_shr(popped[1], right);
        let (result) = uint256_and(shift_right, Uint256(0xFF, 0));
        
        tempvar bitwise_ptr = bitwise_ptr;
        tempvar range_check_ptr = range_check_ptr;
        tempvar result = Uint256(result.low, result.high);
        jmp end;
```

not that we could also use `i.high` in the condition instead of doing the `uint256_lt` which would save one op here but early return less often.